openverse_catalog/dags/common/provider_dag_factory.py

```diff
@@ -96,43 +97,65 @@ def _push_output_paths_wrapper(
     """
     args = args or []
     logger.info("Pushing available store paths to XComs")
-    module = inspect.getmodule(func)
-    stores = {}
-
-    # Stores exist at the module level, so in order to retrieve the output values we
-    # must first pull the stores from the module.
-    for media_type in media_types:
-        if not (store := getattr(module, f"{media_type}_store", None)):
-            continue
-        stores[media_type] = store
-
-    if len(stores) != len(media_types):
-        raise ValueError(
-            f"Expected stores in {module.__name__} were missing: "
-            f"{list(set(media_types) - set(stores))}"
-        )
 
-    for media_type, store in stores.items():
-        logger.info(f"{media_type.capitalize()} store location: {store.output_path}")
-        ti.xcom_push(key=f"{media_type}_tsv", value=store.output_path)
+    if isinstance(ingestion_callable, FunctionType):
+        # Stores exist at the module level, so in order to retrieve the output values we
+        # must first pull the stores from the module.
+        module = inspect.getmodule(ingestion_callable)
+        stores = {}
+        for media_type in media_types:
+            if not (store := getattr(module, f"{media_type}_store", None)):
+                continue
+            stores[media_type] = store
```

Comment:
Oof, this walrus operator hurts my brain, with `store` being used outside the if body :face_with_head_bandage: 

Can we split the assignment and check into two lines? It would make it clearer how store is intended to be used.

```suggestion
            store = getattr(module, f"{media_type}_store", None)
            if not store:
                continue
            stores[media_type] = store
```

Maybe it's just me but any time I see a walrus operator for an if block I read it as a signal that the variable being assigned (especially if it's also being declared) is only relevant to the if body and not the larger function. Maybe this isn't a pythonic way of thinking considering it layers on block scoping that doesn't exist? I'm not sure.

Update: I'm noticing now that this isn't new code in this PR, just moved around, so feel free to ignore.